src/rgw/rgw_data_sync.cc

```diff
@@ -3803,6 +3805,14 @@ class RGWBucketSyncSingleEntryCR : public RGWCoroutine {
         goto done;
       }
       tn->set_flag(RGW_SNS_FLAG_ACTIVE);
+
+      if (sc->env->ostr) {
+	RGWZone* z;
+	if ((z = sc->env->store->svc()->zone->find_zone(sc->source_zone))) {
+	  zone_name = z->name;
+	}
+      }
```

Comment:
can we move this part to the constructor?

Reply:
Done.